### ISSUE TITLE
detektMain: parallel issues

### ISSUE THREAD

**Issue by realdadfish** [detekt/detekt#2629]
I'm working with Detekt 1.8.0 (JDK8, Kotlin 1.3.60) and recently set up type resolution. Now, when I ran `detektMain` with `org.gradle.parallel=true` on a multi-module project, I get weird classloading issues like this:

```
Caused by: java.lang.NoClassDefFoundError: kotlin/ExceptionsKt
        at kotlin.io.CloseableKt.closeFinally(Closeable.kt:58)
        at io.gitlab.arturbosch.detekt.cli.runners.Runner.execute(Runner.kt:30)
        at io.gitlab.arturbosch.detekt.invoke.DefaultCliInvoker.invokeCli(DetektInvoker.kt:51)
        ... 89 more
Caused by: java.lang.ClassNotFoundException: kotlin.ExceptionsKt
        ... 92 more
```

The class that it fails to find is different from time to time, on our CI for example it fails to find ` org/jetbrains/kotlin/cfg/ControlFlowBuilder$PredefinedOperation`. A non-parallel execution of `detektMain` succeeds.

**Comment by margelatu** [comment]
Same here, but running on JDK11.0.2 and Kotlin 1.3.71.

**Comment by schalkms** [comment]
I can confirm. However, I couldn't reproduce it when disabling type resolution.

**Comment by arturbosch** [comment]
@schalkms you could reproduce this in the detekt repo? I couldn't.

**Comment by schalkms** [comment]
I couldn't reproduce it with the detekt repo.

**Comment by realdadfish** [comment]
Hrm... so is an `-Dorg.gradle.parallel=false` execution the only thing we can do at the moment? On CI this wouldn't be a huge issue, because `detekt` runs in parallel with other steps, but locally `detekt` is run as part of a bigger Gradle task and if I disable parallel execution, this would slow down the whole thing.

**Comment by realdadfish** [comment]
So, is this issue somehow fixed magically? Shall I retry with a newer version?

**Comment by schalkms** [comment]
I don’t think so. Can you retest it with your project/setup? @realdadfish

**Comment by realdadfish** [comment]
The issue doesn't seem to pop up any longer. Might have to do with the Gradle upgrade we did in the meantime; we've been on Gradle 5.6.x previously, now we're on Gradle 6.3.x.

**Comment by arturbosch** [comment]
That's good to hear.
Thanks for letting us know.

**Comment by hvandenesker** [comment]
I have the same experience, Detekt 1.14.1 (JDK 1.8 Kotlin 1.4.10 Gradle 6.6.1) 
Gradle in parallel failes, and if not then it is fine.
Depending on the versions you pick (gradle/kotlin/gradle) I get strange missing classes.. for the above combination:

~~~
Caused by: java.lang.NoClassDefFoundError: kotlin/io/CloseableKt
        at io.gitlab.arturbosch.detekt.api.internal.YamlConfig$Companion.load(YamlConfig.kt:69)
        at io.gitlab.arturbosch.detekt.api.internal.YamlConfig$Companion.loadResource(YamlConfig.kt:62)
        at io.gitlab.arturbosch.detekt.core.config.DefaultConfig.newInstance(DefaultConfig.kt:12)
        at io.gitlab.arturbosch.detekt.core.config.ConfigurationsKt.loadConfiguration(Configurations.kt:23)
        at io.gitlab.arturbosch.detekt.core.tooling.ProcessingSpecSettingsBridgeKt$withSettings$configuration$1.invoke(ProcessingSpecSettingsBridge.kt:15)
        at io.gitlab.arturbosch.detekt.core.tooling.ProcessingSpecSettingsBridgeKt$withSettings$configuration$1.invoke(ProcessingSpecSettingsBridge.kt)
~~~

**Comment by kdubb** [comment]
Same here, Detekt 1.14.1 (Kotlin 1.3.72 Gradle 6.7)

When parallel is enabled we get:

    Caused by: java.lang.NoClassDefFoundError: kotlin/io/CloseableKt
            at io.gitlab.arturbosch.detekt.api.internal.YamlConfig$Companion.load(YamlConfig.kt:69)
            at io.gitlab.arturbosch.detekt.api.internal.YamlConfig$Companion.load(YamlConfig.kt:53)
            at io.gitlab.arturbosch.detekt.core.config.ConfigurationsKt.parsePathConfig(Configurations.kt:43)

**Comment by kdubb** [comment]
@arturbosch Should a new issue be opened or re-open this one?

**Comment by arturbosch** [comment]
Please open a new issue, thx.
